src/generate/renders/generated/CliCommands.ts

```diff
@@ -4,313 +4,215 @@
+                    item['propertiesString'][prop] = item[prop];
+                }
+                if (prop === 'resourceType' && !isNullOrUndefined(item[prop])) {
+                    importProfile = true;
+                }
+            });
+            if (
+                !isNullOrUndefined(item['operationTmplName']) &&
+                item['operationTmplName'].length >
+                    CodeGenConstants.PYLINT_MAX_OPERATION_TEMPLATE_LENGTH
+            ) {
+                lineTooLong = true;
+            }
+            if (needWaitCommand && showCustomFunctionName !== '') {
+                item['needWaitCommand'] = true;
+                item['showCustomFunctionName'] = showCustomFunctionName;
+                needWaitCommand = false;
+                showCustomFunctionName = '';
+            } else if (needWaitCommand) {
+                needWaitCommand = false;
+            } else if (showCustomFunctionName !== '') {
+                showCustomFunctionName = '';
+            }
+            return item;
+        };
+
+        const commandConverter = (item) => {
+            item['propertiesString'] = {};
+            extraProperties.forEach((prop) => {
+                if (!isNullOrUndefined(item[prop])) {
+                    item['propertiesString'][prop] = pythonString(item[prop]);
+                }
+            });
+            extraNonStringProperties.forEach((prop) => {
+                if (!isNullOrUndefined(item[prop])) {
+                    item['propertiesString'][prop] = item[prop];
+                }
+                if (prop === 'resourceType' && !isNullOrUndefined(item[prop])) {
+                    importProfile = true;
+                }
+            });
+            if (item['isLongRun'] && showCustomFunctionName !== '') {
+                item['propertiesString']['supports_no_wait'] = 'True';
+                needWaitCommand = true;
+            }
+            if (item['methodName'] === 'delete') {
+                item['propertiesString']['confirmation'] = 'True';
+            }
+            if (item['methodName'] === 'show') {
+                showCustomFunctionName = item['functionName'];
+            }
+            if (item['needGeneric'] && !isNullOrUndefined(item['genericSetterArgName'])) {
+                item['propertiesString']['custom_func_name'] = pythonString(item['functionName']);
+                const setterName = item['genericSetterArgName'];
+                if (setterName && setterName !== '' && setterName !== 'parameters') {
+                    item['propertiesString']['setter_arg_name'] = pythonString(setterName);
+                }
+                if (item['isLongRun'] && !AzConfiguration.getValue(CodeGenConstants.sdkTrack1)) {
+                    item['propertiesString']['setter_name'] = "'begin_create_or_update'";
+                }
+            }
+            return item;
+        };
 
         const inputProperties: Map<CodeModelTypes, RenderInput> = new Map<
```

Comment:
Can we move this out of this function and add them as the  class member?

Reply:
I changed all the converter into class memeber function. BTW, the inputProperties is a data structure definition not a function